### PR TITLE
Update LTS documentation

### PR DESCRIPTION
**Pull request by nashif**
- doc: LTS release docs
- doc: master -> main branch
- doc: more LTS details

### CODE REVIEW

File: doc/development_process/release_process.rst [hunk 1]
```diff
@@ -151,15 +151,90 @@ The following syntax should be used for releases and tags in Git:
...
-An LTS release is made every 2 years and is branched and maintained
-independently from the mainline tree.
+An LTS release is defined as:
+- **Product focused**
+- **Compatible with New Hardware**: We will make point releases throughout the development
+  cycle to provide functional support for new hardware.(FIXME)
+- **Extended Stabilisation period**: Allow for more testing and bug fixing
+- **Stable APIs**
+- **Quality Managed**
+- **Long Term**: Maintained for an extended period of time (at least 2.5 years)
```
Comment (galak) [inline doc/development_process/release_process.rst:165]: do we add a bullet here about overlapping LTS release support?

File: doc/development_process/release_process.rst [hunk 2]
```diff
@@ -151,15 +151,90 @@ The following syntax should be used for releases and tags in Git:
...
+
+(Still WIP)
+
+- API Freeze (LTS - 2)
+  - All stable APIs need to be frozen in 2.6 (release). APIs can extended but not modified.
+    - Device Drivers (i2c.h, spi.h)...
+    - Kernel (k_*):
+    - OS services (logging,debugging, ..)
+    - DTS: API and bindings stability
+    - Kconfig
```
Comment (galak) [inline doc/development_process/release_process.rst:206]: add deprecation removal in LTS-2

File: doc/development_process/release_process.rst [hunk 3]
```diff
@@ -151,15 +151,90 @@ The following syntax should be used for releases and tags in Git:
...
+- Compliance with pubished coding guidelines, style guides and naming
+conventions and documentation of deviations.
+- Regular static analysis on the complete tree using available commercial and
+open-source tools and documentation of deviations and false postives.
+- Documented components and APIS
+- Requirement Catalog
+- Verification Plans
+- Verification Reports
+- Coverage Reports
+- Requirements Traceability Matrix (RTM)
```
Comment (galak) [inline doc/development_process/release_process.rst:229]: Anything about license/SPDX manifest?

**Review by jhedberg — APPROVED** [review]

**Comment by nashif** [comment]
@MaureenHelm  @ioannisg  @carlescufi  can you please review?

File: doc/development_process/release_process.rst [hunk 4]
```diff
@@ -151,15 +151,105 @@ The following syntax should be used for releases and tags in Git:
...
+
+Product Focused
++++++++++++++++
+
+Zephyr LTS is the recommended release for product makers with an extended
+support and maintenance which includes general stability and bug fixes,
+security fixes.
+
+An LTS includes both mature and new features. API and feature maturity is
+docmented and tracked. The footprint and scope of mature and stable APIs expands
```
Comment (ioannisg) [inline doc/development_process/release_process.rst:176]: documented

File: doc/development_process/release_process.rst [hunk 5]
```diff
@@ -151,15 +151,105 @@ The following syntax should be used for releases and tags in Git:
...
+as we move from one LTS to the next given users access to bleading edge features
```
Comment (ioannisg) [inline doc/development_process/release_process.rst:177]: giving

File: doc/development_process/release_process.rst [hunk 6]
```diff
@@ -151,15 +151,105 @@ The following syntax should be used for releases and tags in Git:
...
+and new hardware while keeping a stable foundation that evolves over time.
+
+Extended Stabilisation Period
++++++++++++++++++++++++++++++
+
+Zephyr LTS development cycle differs from regular releases and has an extended
+stabilization period. Feature freeze of regular releases happens 3-4 weeks
+before the scheduled release date. The stabilisation period for LTS is extended
+by 3 weeks with the feature freeze happening 6-7 weeks before the anticipated
```
Comment (ioannisg) [inline doc/development_process/release_process.rst:186]: happening -> occurring

File: doc/development_process/release_process.rst [hunk 7]
```diff
@@ -151,15 +151,105 @@ The following syntax should be used for releases and tags in Git:
...
+release date. The time between code freeze and release date is extended in this case.
+
+Stable APIs
+++++++++++++
```
Comment (ioannisg) [inline doc/development_process/release_process.rst:190]: extra +

File: doc/development_process/release_process.rst [hunk 8]
```diff
@@ -151,15 +151,105 @@ The following syntax should be used for releases and tags in Git:
...
+
+Zephyr LTS provides a stable and long-lived foundation for developing
+products. To guarantee stability of the APIs and the implemementation of such
```
Comment (ioannisg) [inline doc/development_process/release_process.rst:193]: implementation

File: doc/development_process/release_process.rst [hunk 9]
```diff
@@ -151,15 +151,105 @@ The following syntax should be used for releases and tags in Git:
...
+APIs, it is required that any release software that makes the core of the OS
```
Comment (ioannisg) [inline doc/development_process/release_process.rst:194]: remove comma

File: doc/development_process/release_process.rst [hunk 10]
```diff
@@ -151,15 +151,105 @@ The following syntax should be used for releases and tags in Git:
...
+went through the Zephyr API lifecycle and stabilised over at least 2 releases.
+This guarantees that we release many of the highlighted and core feature with
```
Comment (ioannisg) [inline doc/development_process/release_process.rst:196]: features

Also the syntax in the last sentence is not clear to me; what is "implementation supports by stable APIs"?

File: doc/development_process/release_process.rst [hunk 11]
```diff
@@ -151,15 +151,105 @@ The following syntax should be used for releases and tags in Git:
...
+mature and well-established implementations support by stable APIs that are
+supported during the lifetime of the release LTS.
+
+- API Freeze (LTS - 2)
+
+  - All stable APIs need to be frozen 2 release before an LTS. APIs can extended
```
Comment (ioannisg) [inline doc/development_process/release_process.rst:202]: releases

Comment (ioannisg) [reply doc/development_process/release_process.rst:202]: be extended

File: doc/development_process/release_process.rst [hunk 12]
```diff
@@ -151,15 +151,105 @@ The following syntax should be used for releases and tags in Git:
...
+    with additional features, but the core implementation is not modified. This
+    is valid for the following subsystems for example:
+
+    - Device Drivers (i2c.h, spi.h)...
+    - Kernel (k_*):
+    - OS services (logging,debugging, ..)
+    - DTS: API and bindings stability
+    - Kconfig
+
+  - New APIs for experiental features can be added at any time as long as they
```
Comment (ioannisg) [inline doc/development_process/release_process.rst:212]: experimental

File: doc/development_process/release_process.rst [hunk 13]
```diff
@@ -151,15 +151,105 @@ The following syntax should be used for releases and tags in Git:
...
+    - Kernel + Base OS
+    - Additional advertised LTS features
+
+  - Auxiliary features on top of and/or extending the base OS and advertised LTS features
+    can be added at any time and should be marked as experimental if applicable
+
+Quality Driven Process
+++++++++++++++++++++++
+
+The Zephyr project follows industry standards and processes with goal of
```
Comment (ioannisg) [inline doc/development_process/release_process.rst:226]: the goal of

File: doc/development_process/release_process.rst [hunk 14]
```diff
@@ -151,15 +151,105 @@ The following syntax should be used for releases and tags in Git:
...
+providing a quality oritented releases. This is achieved by providing the
```
Comment (ioannisg) [inline doc/development_process/release_process.rst:227]: oriented

File: doc/development_process/release_process.rst [hunk 15]
```diff
@@ -151,15 +151,105 @@ The following syntax should be used for releases and tags in Git:
...
+following products to track progress, integrity and quality of the software
+components provided by the project:
+
+- Compliance with pubished coding guidelines, style guides and naming
+  conventions and documentation of deviations.
+- Regular static analysis on the complete tree using available commercial and
+  open-source tools and documentation of deviations and false postives.
```
Comment (ioannisg) [inline doc/development_process/release_process.rst:234]: positives

File: doc/development_process/release_process.rst [hunk 16]
```diff
@@ -151,15 +151,105 @@ The following syntax should be used for releases and tags in Git:
...
+- Documented components and APIS
+- Requirement Catalog
```
Comment (ioannisg) [inline doc/development_process/release_process.rst:236]: Requirements

File: doc/development_process/release_process.rst [hunk 17]
```diff
@@ -151,15 +151,105 @@ The following syntax should be used for releases and tags in Git:
...
+- SPDX License Reports
+
+Each release is created with the above products to document the quality and the
+state of the software when it was released.
+
+Long Term Support and Maintenance
+++++++++++++++++++++++++++++++++++
+
+A Zephyr LTS release is published every 2 years and is branched and maintained
+independently from the mainline tree for at least 2.5 years after it was
```
Comment (ioannisg) [inline doc/development_process/release_process.rst:250]: main tree, sounds better, and will correspond also to the main branch

**Review by ioannisg — CHANGES_REQUESTED** [review]

**Review by ioannisg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
LGTM

**Pull request merged**